sgt-back-end/index.js

```diff
@@ -0,0 +1,135 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+const parseJson = express.json();
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:lfz@localhost/studentGradeTable'
+});
+
+app.use(parseJson);
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const sql =
+    `select *
+  from "grades"`;
+
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error ocurred'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  const newGrade = req.body;
+  if (!Number.isInteger(newGrade.grade) ||
+    typeof newGrade.name !== 'string' ||
+    typeof newGrade.course !== 'string') {
+    res.status(400).json({
+      error: 'content requirement is not met, gradeId must be a positive integer, name and course must be text'
+    });
+    return;
+  }
+  const sql = `
+  insert into "grades" ("name","course","grade")
+  values($1,$2,$3)`;
+
+  const params = [newGrade.name, newGrade.course, newGrade.grade];
+  db.query(sql, params)
+    .then(result => {
+      res.json(params);
```

Comment:
The client should receive the inserted grade, so you'll need a `returning` clause in your `insert` statement.

Reply:
The client should not receive an array as noted above.